src/Bicep.LangServer/Handlers/BicepBuildParamsCommandHandler.cs

```diff
@@ -0,0 +1,120 @@
+        {
+            string compiledFilePath = PathHelper.ResolveParametersFileOutputPath(bicepParamsFilePath, OutputFormatOption.Json);
+            string compiledFile = Path.GetFileName(compiledFilePath);
+
+            // If the template exists and contains bicep generator metadata, we can go ahead and replace the file.
+            // If not, we'll fail the generate params.
+            if (File.Exists(compiledFilePath) && !TemplateIsParametersFile(File.ReadAllText(compiledFilePath)))
+            {
+                return "Building parameters file failed. The file \"" + compiledFile + "\" already exists but does not contain the schema for a parameters file. If overwriting the file is intended, delete it manually and retry the Generate Parameters command.";
+            }
+
+            var compilation = await new CompilationHelper(bicepCompiler, compilationManager).GetRefreshedCompilation(documentUri);
+            var fileUri = documentUri.ToUri();
+
+            var diagnosticsByFile = compilation.GetAllDiagnosticsByBicepFile().FirstOrDefault(x => x.Key.FileUri == fileUri);
+
+            if (diagnosticsByFile.Value.Any(x => x.Level == DiagnosticLevel.Error))
+            {
+                return "Building parameters file failed. Please fix below errors:\n" + DiagnosticsHelper.GetDiagnosticsMessage(diagnosticsByFile);
+            }
+
+            var existingContent = File.Exists(compiledFilePath) ? File.ReadAllText(compiledFilePath) : string.Empty;
+
+            var paramsSemanticModel = compilation.GetEntrypointSemanticModel();
+
+            if (paramsSemanticModel.Root.TryGetBicepFileSemanticModelViaUsing(out var bicepSemanticModel, out _))
+            {
+                var bicepFileUsingPathUri = bicepSemanticModel.Root.FileUri;
+
+                static string DefaultOutputPath(string path) => PathHelper.GetDefaultBuildOutputPath(path);
+                var paramsOutputPath = PathHelper.ResolveDefaultOutputPath(bicepParamsFilePath, null, compiledFilePath, DefaultOutputPath);
+
+                var model = compilation.GetEntrypointSemanticModel();
+                var emitter = new TemplateEmitter(model);
+
+                using var fileStream = new FileStream(compiledFilePath, FileMode.Create, FileAccess.Write);
+
+                var result = new ParametersEmitter(model).Emit(fileStream);
+            }
+
+            return "Building parameters file succeeded. Processed file " + compiledFile;
+        }
+
+        // Returns true if the template contains the parameters file schema, false otherwise
+        public bool TemplateIsParametersFile(string template)
+        {
+            try
+            {
+                if (!string.IsNullOrEmpty(template))
+                {
+                    JToken jtoken = template.FromJson<JToken>();
+                    var schema = jtoken.SelectToken("$schema")?.ToString();
+
+                    if (schema != null && schema.ContainsInsensitively("deploymentParameters.json"))
```

Comment:
I think it may be better to use a regex to match `https://schema.management.azure.com/schemas/<api-version>/deploymentParameters.json#` here.

Reply:
I'm using the same structure as the [BicepGenerateParamsCommandHandler.cs#L101](https://github.com/kaizentm/bicep/blob/780b1d84cfb555f6b7dd523b0de2bbf943d2307e/src/Bicep.LangServer/Handlers/BicepGenerateParamsCommandHandler.cs#L101)

This type of check could be somewhere else, too. I think it could be a follow-up PR to improve schema checks.

Reply:
A follow-up PR makes sense. Let's leave this as is then.

Reply:
it could be a simple check, like;

```
var pattern = @"https?://schema\.management\.azure\.com/schemas/.*/deploymentParameters\.json";

if (schema != null && Regex.IsMatch(schema, pattern, RegexOptions.IgnoreCase))
{
    return true;
}
```

le me create a separate PR for that 👍

Reply:
Sounds good!

Reply:
Btw, I'd recommend using regex source generator when implementing this. See: https://learn.microsoft.com/en-us/dotnet/standard/base-types/regular-expression-source-generators.